### PR TITLE
Fix/use index

### PR DESCRIPTION
**Pull request by jgfidelis**
https://mini--delivery.myvtex.com/
https://mini--storecomponents.myvtex.com/

related to
https://github.com/vtex-apps/product-summary/pull/134
vtex-apps/store-resources#15

#### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,11 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Changed
+- Strigify and parse local state order form to prevent bugs in apollo link state.
```
Comment (brunojdo) [inline CHANGELOG.md]: Stringify

File: react/index.js [hunk 2]
```diff
@@ -318,8 +306,8 @@ const withLinkStateMinicartQuery = graphql(fullMinicartQuery, {
   options: () => ({ ssr: false }),
   props: ({ data: { minicart } }) => ({
     linkState: {
-      minicartItems: minicart && minicart.items,
-      orderForm: minicart && minicart.orderForm,
+      minicartItems: minicart && JSON.parse(minicart.items),
+      orderForm: minicart && JSON.parse(minicart.orderForm),
```
Comment (lbebber) [inline react/index.js:310]: are you absolutely sure these values are going to be valid JSON strings? `JSON.parse` throws errors when it fails

File: react/localState/index.js [hunk 3]
```diff
@@ -23,7 +23,9 @@ export default function(client) {
       minicart: { orderForm: outdatedOrderForm },
     } = cache.readQuery({ query: minicartOrderFormQuery })
 
-    const orderForm = mergeDeepRight(outdatedOrderForm, updatedOrderForm)
+    const orderForm = JSON.stringify(
```
Comment (lbebber) [inline react/localState/index.js:26]: ditto

File: react/localState/index.js [hunk 4]
```diff
@@ -40,51 +42,38 @@ export default function(client) {
...
-          const { index, item } = indexedItem
-          if (index !== -1) {
-            prevItems[index] = item
-          } else {
-            newItems.push(item)
-          }
-        }
-
+        const newItems = items.map(item => mapToMinicartItem({ ...item, upToDate: false }))
+        const writeItems = [...JSON.parse(prevItems), ...newItems]
```
Comment (lbebber) [inline react/localState/index.js:46]: ditto

Comment (lbebber) [reply react/localState/index.js:46]: etc etc

File: react/index.js (hunk 2 above)
Comment (jgfidelis) [reply react/index.js:310]: I am.

**Review by lbebber — APPROVED** [review]

**Review by lucis — APPROVED** [review]

**Pull request merged**